### ISSUE TITLE
Documentation for GitHub Codespaces

### ISSUE THREAD

**Issue by samruddhikhandale** [mastodon/mastodon#23346]
Hi 👋 

I am Samruddhi, a Software Engineer for GitHub, working with the [GitHub Codespaces](https://github.com/features/codespaces) team, and a maintainer for the [devcontainers](https://github.com/devcontainers) org 👩‍💻

I was looking at the [mastodon/mastodon](https://github.com/mastodon/mastodon) repo which has a pretty cool [devcontainer configuration](https://github.com/mastodon/mastodon/tree/main/.devcontainer) setup ✨ 

[GitHub Codespaces](https://docs.github.com/en/codespaces/overview) is a cloud-based development environment that uses your dev container to provide great contributor and maintainer experiences for all users of your project 🪄

🙅‍♀️ No one would ever have to worry about setting up their local machines, adding tools/runtimes for a specific project (which most frequently contradicts with their current environment) and spend hours before even starting their actual work 🙅‍♀️

Would you be willing to accept improvements to your current [devcontainer config](https://github.com/mastodon/mastodon/tree/main/.devcontainer) and a documentation to improve the GitHub Codespaces experience? 💁‍♀️

Thanks,
Samruddhi

// cc @craiglpeters

**Comment by samruddhikhandale** [comment]
Hi 👋 

Tagging for more visibility on this issue.

@ykzts @ClearlyClaire @Gargron Thanks!

**Comment by samruddhikhandale** [comment]
Opened a PR - https://github.com/mastodon/mastodon/pull/23872

**Comment by dj-sf** [comment]
Hey @samruddhikhandale .  I've been working on mastodon with github codespaces for a few weeks and overall I have been very happy with them.  I was in a situation where I basically only had a chromebook to work on a time-sensitive project while traveling for over 2 weeks.  Apart from a few issues with image building and rebuilding, it was a great experience and I loved how I was able to pick right back up where I left off without having to reconfigure everything on my better computer.

However the images themselves that Mastodon is using have a lot of of inconsistencies and issues that make it difficult to get set up and configured quickly.  Mostly, there are multiple separate installations of both `rvm`, and `rbenv`, and also a plain system Ruby instance.  These are all referenced at different times in different scripts, and they are added to the `$PATH` separately in both .zshrc and .bashrc and a few other places, in different orders depending on which script you run. They are all technically the same version (3.2.2) but they are all over the place and result in bundle and gem installations being scattered across the system. 

This resulted in a lot of headaches and confusion and it took me almost 2 weeks to figure this all out to the point where I had a stable development environment with modern tooling configured.

Therefore I'd like to propose that we start prioritizing improvements to the Docker images in our devcontainer setup.   [I've got an image in the works now](https://hub.docker.com/repository/docker/djsf/mastodon/general)... it isn't super great at the moment because it only really works well with the [post-create  script ](https://github.com/djsf-kobayashi/mastodon-solid-poc/blob/solidon-working-branch/.devcontainer/post-create.sh) I've modified in my own working branch... (and it has a few other issues too) but it basically is designed so that `rbenv` is unambiguously the ruby version manager that one should use.  

I considered keeping `rvm` too but decided on `rbenv` because it's more lightweight and possibly more appropriate for a docker container designed with a single purpose in mind.

This is a start that I'd possibly like to build on moving forward.  Would be great to lower the barrier to entry for newcomers to start tinkering with it.  I'm open to any and all ideas!
